src/ripple/consensus/Validations.h

```diff
@@ -723,15 +731,16 @@ class Validations
             {
```

Comment:
Thanks, that is a good idea. Changed and added some unit tests. 
BTW, I think `Validations::expire()` is called every minute, instead of every ledger close.

Reply:
You're partly right about how often `Validations::expire()` is called.  I put a print at the top of `Validations::expire()`. On my system I see that print pop out about every 10 or 11 seconds, not every four.  So it's certainly not happening with every ledger close.  Maybe `Validations::expire()` is called from more places than you realize?

Reply:
Scott and I had some discussion. To summarize, `Validations::expire()` is called in `Application::doSweep()`. Scott found out that the call frequency depended on node size. My node size is `medium`, so `doSweep()` happens every minute, and Scott's is every 10 seconds because his node size is `tiny`. 

Note that the time guard Scott proposed still works in all cases. 